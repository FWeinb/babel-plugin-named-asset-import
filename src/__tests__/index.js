import { transform } from 'babel-core'
import pluginTester from 'babel-plugin-tester'
import plugin from '../'


it('should add import to webpack loaders', () => {
    const input = `
        import { ReactComponent, url } from 'icon.svg'
    `
    const { code } = transform(input, { plugins: [plugin] });
    expect(code).toMatchSnapshot();
});

it('should throw if there was an invalid asset name', () => {
    const input = `
    import { thisIsNotExported } from 'icon.svg'
    `
    expect(() => {
        const { code } = transform(input, { plugins: [plugin] });
    }).toThrow(/url, raw, ReactComponent/);
})

it('should retained renamed import', () => {
    const input = `
        import { ReactComponent as Icon } from 'icon.svg'
    `
    const { code } = transform(input, { plugins: [plugin] });
    expect(code).toMatchSnapshot();
});

it('should leave default import intact for now', () => {
    const input = `
        import Test from 'icon.svg'
    `
    const { code } = transform(input, { plugins: [plugin] });
    expect(code).toMatchSnapshot();
});

it('modules and js files should not be changed', () => {
    const input = `
        import { url, other, key } from 'some-module'
        import { test } from './file.js'
    `
    const { code } = transform(input, { plugins: [plugin] });
    expect(code).toMatchSnapshot();
});