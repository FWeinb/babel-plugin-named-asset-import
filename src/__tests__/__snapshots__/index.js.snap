// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules and js files should not be changed 1`] = `
"
import { url, other, key } from 'some-module';
import { test } from './file.js';"
`;

exports[`should add import to webpack loaders 1`] = `
"import ReactComponent from 'svg-react-loader!icon.svg';
import url from 'url-loader!icon.svg';"
`;

exports[`should leave default import intact for now 1`] = `
"
import Test from 'icon.svg';"
`;

exports[`should retained renamed import 1`] = `"import Icon from 'svg-react-loader!icon.svg';"`;

exports[`should work for markdown and html 1`] = `
"import docHtml from 'html-loader!markdown-loader!doc.md';
import raw from 'file-loader!doc.md';
import otherDocHtml from 'html-loader!otherDoc.html';
import url from 'url-loader!otherDoc.html';"
`;
